Objects/dictobject.c

```diff
@@ -1475,6 +1475,17 @@ _PyDict_SetItem_KnownHash(PyObject *op, PyObject *key, PyObject *value,
     return insertdict(mp, key, hash, value);
 }
 
+// Resize dict if dk_size is too large.
+static int
+after_removed(PyDictObject *mp)
+{
+    if (mp->ma_keys->dk_size > PyDict_MINSIZE &&
```

Comment:
Oh, I hesitated to propose the same test in your previous PR #5297, but I wasn't sure if it was required in practice since MINSIZE=8.

I like the idea of being explicit here, check size > MINSIZE.